pub-critic/src/app/components/GamesList/GamesList.tsx

```diff
@@ -0,0 +1,80 @@
+"use client";
+import { Game } from "api/GamesShared";
+import classes from "./GamesList.module.scss";
+import GameCard from "components/GameCard";
+import { useRef, useState, useEffect } from "react";
+import { getFilteredGames } from "api/GamesApi";
+
+export interface GamesListProps {
+  games: Game[];
+  searchParams?: {
+    search?: string;
+    genre?: number;
+    platform?: number;
+    metacritic?: string;
+    page?: number;
+    pageSize?: number;
+  };
+}
+export const GamesList = ({ games, searchParams }: GamesListProps) => {
+  const list = useRef<HTMLDivElement>(null);
+  const [currentPage, setCurrentPage] = useState<number>(
+    searchParams?.page || 1
+  );
+
+  const [visibleGames, setVisibleGames] = useState<Game[]>(games);
+  const [loading, setLoading] = useState<boolean>(false);
+  const fetchMore = async () => {
+    if (loading) return;
+    setLoading(true);
+    const newGames = await getFilteredGames({
+      search: searchParams?.search as string,
+      genre: searchParams?.genre as number,
+      platform: searchParams?.platform as number,
+      metacritic: searchParams?.metacritic as string,
+      page: currentPage + 1,
+      pageSize: searchParams?.pageSize as number,
+    });
+    if (newGames.results) {
+      setVisibleGames((prev) => [...prev, ...newGames.results]);
+      setCurrentPage((prev) => prev + 1);
+    }
+    setLoading(false);
+  };
+
+  useEffect(() => {
+    if (list.current) {
+      list.current.scrollIntoView({ behavior: "smooth" });
+    }
+    setVisibleGames(games);
+  }, [games]);
+
+  useEffect(() => {
+    if (list.current) {
+      list.current.scrollIntoView({ behavior: "smooth" });
+    }
+  }, []);
+
+  useEffect(() => {
+    window.addEventListener("scroll", handleScroll);
+    return () => window.removeEventListener("scroll", handleScroll);
+  });
```

Comment:
do you miss `handleScroll` dependency here?

Reply:
that would throw this warning - The 'handleScroll' function makes the dependencies of useEffect Hook (at line 71) change on every render. Move it inside the useEffect callback. Alternatively, wrap the definition of 'handleScroll' in its own useCallback() Hook.eslint[react-hooks/exhaustive-deps](https://github.com/facebook/react/issues/14920)